src/catalog/column_catalog.cpp

```diff
@@ -166,171 +170,63 @@ bool ColumnCatalog::DeleteColumn(oid_t table_oid,
-  values.push_back(type::ValueFactory::GetIntegerValue(column_id).Copy());
-
-  auto result_tiles =
-      GetResultWithIndexScan(column_ids, index_offset, values, txn);
-
-  std::string column_name;
-  PL_ASSERT(result_tiles->size() <= 1);  // table_oid & column_offset is unique
-  if (result_tiles->size() != 0) {
-    PL_ASSERT((*result_tiles)[0]->GetTupleCount() <= 1);
-    if ((*result_tiles)[0]->GetTupleCount() != 0) {
-      column_name = (*result_tiles)[0]
-                        ->GetValue(0, 0)
-                        .ToString();  // After projection left 1 column
-    }
-  }
-
-  return column_name;
-}
-
-type::TypeId ColumnCatalog::GetColumnType(oid_t table_oid,
-                                                std::string column_name,
-                                                concurrency::Transaction *txn) {
-  std::vector<oid_t> column_ids({4});  // column_type
-  oid_t index_offset = 0;              // Index of table_oid & column_name
-  std::vector<type::Value> values;
-  values.push_back(type::ValueFactory::GetIntegerValue(table_oid).Copy());
-  values.push_back(
-      type::ValueFactory::GetVarcharValue(column_name, nullptr).Copy());
-
-  auto result_tiles =
-      GetResultWithIndexScan(column_ids, index_offset, values, txn);
+  // delete columns from cache
+  auto table_object =
+      TableCatalog::GetInstance()->GetTableObject(table_oid, txn);
+  table_object->EvictAllColumnObjects();
 
-  type::TypeId column_type = type::TypeId::INVALID;
-  PL_ASSERT(result_tiles->size() <= 1);  // table_oid & column_name is unique
-  if (result_tiles->size() != 0) {
-    PL_ASSERT((*result_tiles)[0]->GetTupleCount() <= 1);
-    if ((*result_tiles)[0]->GetTupleCount() != 0) {
-      column_type =
-          (*result_tiles)[0]
-              ->GetValue(0, 0)
-              .GetAs<type::TypeId>();  // After projection left 1 column
-    }
-  }
-
-  return column_type;
+  return DeleteWithIndexScan(index_offset, values, txn);
 }
 
-type::TypeId ColumnCatalog::GetColumnType(oid_t table_oid,
-                                                oid_t column_id,
-                                                concurrency::Transaction *txn) {
-  std::vector<oid_t> column_ids({4});  // column_type
-  oid_t index_offset = 1;              // Index of table_oid & column_id
+const std::unordered_map<oid_t, std::shared_ptr<ColumnCatalogObject>>
+ColumnCatalog::GetColumnObjects(oid_t table_oid,
+                                concurrency::Transaction *txn) {
+  // try get from cache
+  // auto table_object = txn->catalog_cache.GetTableObject(table_oid);
+  auto table_object =
+      TableCatalog::GetInstance()->GetTableObject(table_oid, txn);
+  PL_ASSERT(table_object && table_object->table_oid == table_oid);
+  auto column_objects = table_object->GetColumnObjects(true);
+  if (column_objects.size() != 0) return column_objects;
+
+  // cache miss, get from pg_attribute
+  std::vector<oid_t> column_ids({0, 1, 2, 3, 4, 5, 6, 7});
```

Comment:
Are these numbers intentionally hard coded. Shouldnt we be defining them earlier on as constants and using them here if we need to?